`flake8: error: unrecognized arguments: --diff` during linting CI test

Our linting check has started failing on recent PRs. ([Example run](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/3535457184/jobs/5933500966)) 

The error given is:

```bash
##[debug]Evaluating condition for step: 'Lint with flake8'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: Lint with flake8
##[debug]Loading inputs
##[debug]Loading env
Run git diff origin/master... | flake8 --diff
##[debug]/usr/bin/bash -e /home/runner/work/_temp/4a22afc9-d7d5-45e4-90ec-ba759be1d28b.sh
usage: flake8 [options] file file ...
flake8: error: unrecognized arguments: --diff
Error: Process completed with exit code 2.
##[debug]Finishing: Lint with flake8
```

Looking at the installed version of `flake8` before/after failing runs shows that automatic upgrade to `flake8=6.0.0` was the cause here. I'll see if I can find an upstream issue to link this to.

I presume we can pin to `flake8<6.0.0`, but I'm curious if `flake8` provides an alternative option to `--diff`... :thinking:

--- a/.github/workflows/lint_code.yml
+++ b/.github/workflows/lint_code.yml
@@ -14,7 +14,7 @@ jobs:
       with:
         python-version: 3.8
     - name: Install dependencies
-      run: pip install flake8
+      run: pip install flake8<6.0.0
     # Note: flake8 picks up project-wide configuration options from 'setup.cfg' in SCT's root directory
     - name: Lint with flake8
       # NB: 'master...branch' is syntactic sugar for 'git diff `git merge-base master branch`..branch'

--- a/.github/workflows/lint_code.yml
+++ b/.github/workflows/lint_code.yml
@@ -14,7 +14,7 @@ jobs:
       with:
         python-version: 3.8
     - name: Install dependencies
-      run: pip install flake8<6.0.0
+      run: pip install 'flake8<6.0.0'
     # Note: flake8 picks up project-wide configuration options from 'setup.cfg' in SCT's root directory
     - name: Lint with flake8
       # NB: 'master...branch' is syntactic sugar for 'git diff `git merge-base master branch`..branch'

